w1thermsensor/core.py

```diff
@@ -404,6 +406,36 @@ def set_precision(self, precision, persist=False):
 
         return True
 
+    def set_offset(self, offset):
+        """
+            Set an offset to be applied to each temperature reading. This is
+            used to tune sensors which report values which are either too high
+            or too low.
+
+            The offset is converted as needed when getting temperatures in
+            other units than C.
```

Comment:
Does it make sense to add a *unit* argument? So that people don't have to care about Celsius?

Reply:
Yes. That would be best. I think we would then need some logic for converting different units into celcius. Right now, I guess `UNIT_FACTORS` only goes the other way?

Should I simply add another dict, similar to  `UNIT_FACTORS` for doing conversions to celcius, or do you see a better way?

Reply:
> I think we need to store the offset internally in C

Yes, I'd convert the offset from whatever given unit in `set_offset` to C and store it as a member (like it is right now)

Reply:
> Should I simply add another dict,

Maybe it's best to only have a single dict with the conversions (from, to) as tuple for the keys and the functions as value?
The existing ones would then be:

```python
UNIT_FACTORS = {
    (DEGREES_C, DEGREES_C): lambda x:x,
    (DEGREES_C, DEGREES_F): lambda x: ...,
    (DEGREES_C, KEVLIN): lambda x: ...,
}
```

What do you think?

Reply:
> > Should I simply add another dict,
> 
> Maybe it's best to only have a single dict with the conversions (from, to) as tuple for the keys and the functions as value?
> The existing ones would then be:
> 
> ```python
> UNIT_FACTORS = {
>     (DEGREES_C, DEGREES_C): lambda x:x,
>     (DEGREES_C, DEGREES_F): lambda x: ...,
>     (DEGREES_C, KEVLIN): lambda x: ...,
> }
> ```
> 
> What do you think?

I think this makes sense. I've pushed an implementation with the tuple approach. I think it turned out well.